routing/mock_test.go

```diff
@@ -302,20 +302,38 @@ func (m *mockControlTower) RegisterAttempt(phash lntypes.Hash,
 	m.Lock()
 	defer m.Unlock()
 
+	// Lookup payment.
+	p, ok := m.payments[phash]
+	if !ok {
+		return channeldb.ErrPaymentNotInitiated
+	}
+
+	var inFlight bool
+	for _, a := range p.attempts {
```

Comment:
Always a tradeoff there, because we could be running tests that mock `RegisterAttempt` calls that return the wrong value - part of why we never caught this bug in the first place. I think that it could actually be worth using a test db in these because there's a lot of state logic in some of the payment control calls? But perhaps in a follow up since there's a lot of refactor-tractoring in this already.